[BOUNTY] Team members

ZAPS0N opened the pull request:
More details:
https://www.vintagestory.at/support/17602/

ApacheTech commented on lib/config.php:
```diff
@@ -8,11 +8,18 @@
-	define("DEBUGUSER", 1);
+	$config["databasehost"] = "127.0.0.1";
+	$config["databaseuser"] = "vsmoddb";
+	$config["databasepassword"] = "vsmoddb";
```
Cleartext passwords uploaded within PR. Move to environment variables, user secrets, or key vault.

ZAPS0N replied:
These are the default credentials from the docker-compose.yml file in the repository. In other scenarios, I would certainly use environment variables.

ApacheTech commented on .devcontainer/docker-compose.yml:
```diff
@@ -33,19 +33,28 @@ services:
+    image: phpmyadmin/phpmyadmin
+    environment:
+      PMA_HOST: db
+      MYSQL_ROOT_PASSWORD: root
```
Cleartext passwords uploaded within PR. Move to environment variables, user secrets, or key vault.

ZAPS0N replied:
As above

tyronx commented:
thanks for the submission! I will begin looking into this now

tyronx commented on templates/edit-mod.tpl:
```diff
@@ -55,125 +59,171 @@
 
+						<select name="newownerid" style="width:300px;">
+							<option value="" selected="selected">--- Select new owner ---</option>
+							{foreach from=$users item=user}
```
this overwrites the self user $user, needs to be renamed to `{foreach from=$users item=ouser}`

tyronx commented:
Got this error when saving team members
![image](https://github.com/user-attachments/assets/c6ffb236-04b1-4fee-878b-d910f00458b8)

tyronx commented:
Lots of warnings when accepting an invitation
![image](https://github.com/user-attachments/assets/f8bab148-97b9-4913-ab90-95a845e5c352)

tyronx commented:
more warnings when offered mod ownership
![image](https://github.com/user-attachments/assets/e3abf2e2-814e-430a-9a50-9f7b761e83b8)

ZAPS0N commented:
Hmm that's weird. I haven't encountered those issues. Does disabling debug mode resolve the problem? I've observed that certain functions, for example, API responses, can be invalid when debug mode is active. What I also noticed is that redirections from the notification controller don't work in debug mode

tyronx commented:
Disabling debug mode just disables display of those warnings, but they'll still be printed to the log file.

ZAPS0N commented:
I'll check this out today. I didn't see these errors the last time I was working on it.

ZAPS0N commented:
Okay, I've fixed some warnings/errors related to saving team members. However, I couldn't reproduce the issues with accepting team invitations or mod ownership. Did you see the support ticket? It details my changes, including the addition of 'teaminvite' and 'modownershiptransfer' ENUM types to the 'type' column in the 'notification' databse table. Have you updated the database accordingly? The structure of the table was also updated in the '000_tables.sql' file.

tyronx commented:
Yes i applied these. Mod ownership issue is caused by this incorrect sql statement in notificatio.php
![image](https://github.com/user-attachments/assets/266de605-4e05-4f88-83cb-6ea3586196de)

ZAPS0N commented:
I made the changes, but I'm not sure why it didn't work on your side, but it worked for me. I don't know if this is important, but I'm running my installation locally using docker and dev containers in Visual Studio Code.

tyronx commented:
Maybe your assetid happens to be identical to the modid and thus accidentally correct. There are still more warnings that are not fixed yet. How shall we proceed?

ZAPS0N commented:
I've addressed the potential causes of the errors. To ensure a clean state and verify the fix, please truncate the 'teammembers' table.

tyronx commented:
oh yes. this works now!

tyronx commented:
Can you resolve the merge conflicts please still, then we can merge it

ZAPS0N commented:
I've fixed the merge conflicts

tyronx commented:
thanks!

tyronx commented:
Uggghhh, problem: You use JSON_ARRAYAGG, which is only supported in newer versions of MariaDB. The moddb server only has mariadb 10.2 installed and upgrading would amount to 10-20 hours of work. Can you please refactor this to not use this method.

ZAPS0N commented:
Sure, I created new pull request with fix: https://github.com/anegostudios/vsmoddb/pull/43
